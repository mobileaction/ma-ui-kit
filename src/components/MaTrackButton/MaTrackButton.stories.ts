import MaTrackButton from './MaTrackButton.vue';

export default {
  title: 'MaTrackButton',

  parameters: {
    component: MaTrackButton,
  },
};

export const Orange = () => ({
  components: { MaTrackButton },
  template: `
      <div class="m-2">
        <ma-track-button keyword="keyword" color="orange"/>
        <ma-track-button keyword="long keyword" color="orange" loading/>
        <ma-track-button keyword="keyword" color="orange" tracked/>
        <ma-track-button keyword="keyword" disabled/>
        <ma-track-button keyword="keyword" color="orange" size="s"/>
        <ma-track-button keyword="keyword" size="s" disabled/>
      </div>
    `,
});
Orange.storyName = 'orange';

export const LightGreen = () => ({
  components: { MaTrackButton },
  template: `
      <div class="m-2">
        <ma-track-button keyword="keyword" color="light-green"/>
        <ma-track-button keyword="long keyword" color="light-green" loading/>
        <ma-track-button keyword="keyword" color="light-green" tracked/>
        <ma-track-button keyword="keyword" disabled/>
        <ma-track-button keyword="keyword" color="light-green" size="s"/>
        <ma-track-button keyword="keyword" size="s" disabled/>
      </div>
    `,
});

LightGreen.storyName = 'light-green';

export const LightBlue = () => ({
  components: { MaTrackButton },
  template: `
    <div class="m-2">
    <ma-track-button keyword="keyword" color="light-blue"/>
    <ma-track-button keyword="long keyword" color="light-blue" loading/>
    <ma-track-button keyword="keyword" color="light-blue" tracked/>
    <ma-track-button keyword="keyword" disabled/>
    <ma-track-button keyword="keyword" color="light-blue" size="s"/>
    <ma-track-button keyword="keyword" size="s" disabled/>
    </div>
    `,
});
LightBlue.storyName = 'light-blue';

export const DrakBlue = () => ({
  components: { MaTrackButton },
  template: `
    <div class="m-2">
    <ma-track-button keyword="keyword" color="dark-blue"/>
    <ma-track-button keyword="long keyword" color="dark-blue" loading/>
    <ma-track-button keyword="keyword" color="dark-blue" tracked/>
    <ma-track-button keyword="keyword" disabled/>
    <ma-track-button keyword="keyword" color="dark-blue" size="s"/>
    <ma-track-button keyword="keyword" size="s" disabled/>
    </div>
    `,
});
DrakBlue.storyName = 'dark-blue';

export const Grey = () => ({
  components: { MaTrackButton },
  template: `
    <div class="m-2">
    <ma-track-button keyword="keyword" color="grey"/>
    <ma-track-button keyword="long keyword" color="grey" loading/>
    <ma-track-button keyword="keyword" color="grey" tracked/>
    <ma-track-button keyword="keyword" disabled/>
    <ma-track-button keyword="keyword" color="grey" size="s"/>
    <ma-track-button keyword="keyword" size="s" disabled/>
    </div>
    `,
});
Grey.storyName = 'grey';

export const LightRed = () => ({
  components: { MaTrackButton },
  template: `
    <div class="m-2">
    <ma-track-button keyword="keyword" color="light-red"/>
    <ma-track-button keyword="loading" color="light-red"loading/>
    <ma-track-button keyword="keyword" color="light-red" tracked/>
    <ma-track-button keyword="keyword" disabled/>
    <ma-track-button keyword="keyword" color="light-red" size="s"/>
    <ma-track-button keyword="keyword" size="s" disabled/>
    </div>
    `,
});
LightRed.storyName = 'light-red';

export const Loading = () => ({
  components: { MaTrackButton },
  template: `
    <div class="m-p-10">
    <ma-track-button icon="spinner" spin> Button </ma-track-button>
    </div>
    `,
});
Loading.storyName = 'loading';

export const Disabled = () => ({
  components: { MaTrackButton },
  template: `
    <div class="m-p-10">
    <ma-track-button left-icon="plus" disabled>Button </ma-track-button>
    </div>
    `,
});

Disabled.storyName = 'disabled';
