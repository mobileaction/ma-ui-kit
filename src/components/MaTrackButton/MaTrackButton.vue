<template>
    <a-button
        v-bind="$attrs"
        class="ma-button"
        :class="classNames"
        :disabled="disabled"
        v-on="$listeners"
        @click="$emit('click',$event)"
    >
        <span class="ma-btn-content items-center">
            <span class="flex flex-wrap content-center mr-1">
                <span>
                    <span v-if="!loading">
                        <ma-icon v-if="tracked" icon="check" class="ma-btn-icon"/>
                        <ma-icon v-else icon="plus" class="ma-btn-icon"/>
                    </span>
                    <span v-else-if="loading">
                        <ma-icon icon="spinner" spin/>
                    </span>
                </span>
            </span>
            <span class="text-left leading-tight">
                <p>
                    {{ keyword }}
                </p>
                <span v-if="translatedKeyword" class="translatedKw text-xs">
                    (en) {{ translatedKeyword }}
                </span>
            </span>
        </span>
    </a-button>
</template>

<script>
    import Button from 'ant-design-vue/lib/button/index.js';
    import MaIcon from '../MaIcon/MaIcon.vue';

    import { library } from '@fortawesome/fontawesome-svg-core';
    import { faTrashAlt, faSpinner } from '@fortawesome/free-solid-svg-icons';
    library.add(faTrashAlt, faSpinner);
    export default {
        name: 'ma-track-button',
        components: {
            AButton: Button,
            MaIcon,
        },
        props: {
            trackAll: { type: Boolean },
            color: { type: String, default: 'default' },
            size: { type: String, default: 'm' },
            loading: { type: Boolean, default: false },
            disabled: { type: Boolean, default: false },
            tracked: { type: Boolean, default: false },
            keyword: { type: String, required: true },
            translatedKeyword: { type: String, default: '' },
        },
        computed: {
            classNames () {
                return {
                    [`-tracked-${this.color}`]: this.tracked,
                    [`-size-${this.size}`]: true,
                    [`-${this.color}`]: true,
                };
            },
        },
    };
</script>

<style lang="scss" scoped>
  @import '~ant-design-vue/lib/button/style/index.css';
    .ant-btn:hover:not([disabled]),
    .ant-btn:focus {
      background-color: #FAFAFA;
      border-color: #656565;
      color: #656565;
    }
    .ma-button {
      align-items: center;
      border-radius: 2px;
      min-height: 33px;
      height: auto;
      font-size:13px;
      padding: 5px 10px;
      .ma-btn-content {
        display: flex;
      }
      &.-default {
        &.-tracked-default {
          background-color: #f3f3f3;
          border-color: #d9d8d8;
          color: #aaaaaa;
          &:hover {
            &:not([disabled]){
              border-color: #656565;
              color: #656565;
            }
          }
        }
      }
      &.-orange {
        &:not([disabled]){
          color: #FF9762;
          border-color: #FF9762;
        }
        &:focus,
        &:hover {
          border-color: #FF9762;
          color: #FF9762;
          background-color: #FFF7F1;
        }
      }
      &.-tracked-orange {
        border-color: #FF9762;
        color: #FF9762;
        background-color: #FFF7F1;
      }

      &.-light-green {
        &:not([disabled]){
          color: #24CEB8;
          border-color: #23CEB8;
        }
        &:focus,
        &:hover {
          background-color: #EAFFFD;
          color: #23CEB8;
        }
      }
      &.-tracked-light-green {
        background-color: #EAFFFD;
        color: #23CEB8;
      }

      &.-light-blue {
        &:not([disabled]){
          color: #6F8CC5;
          border-color:#6F8CC5;
        }
        &:focus,
        &:hover {
          color: #6F8CC5;
          background-color: #E4EDFE;
        }
      }
      &.-tracked-light-blue {
        background-color: #E4EDFE;
      }

      &.-dark-blue {
        &:not([disabled]){
          color: #3F83A3;
          border-color:#3F83A3;
        }
        &:focus,
        &:hover {
          background-color: #DFEDF3;
        }
      }
      &.-tracked-dark-blue {
        background-color: #DFEDF3;
      }

      &.-grey {
        &:not([disabled]){
          color: #435160;
          border-color: #435160;
        }
        &:focus,
        &:hover {
          background-color: #F1F1F3;
        }
      }
      &.-tracked-grey {
        background-color: #F1F1F3;
      }

      &.-light-red {
        &:not([disabled]){
          color: #FF7262;
          border-color: #FF7262;
        }
        &:focus,
        &:hover {
          background-color: #FCE4E2;
        }
      }
      &.-tracked-light-red {
        background-color: #FCE4E2;
      }

      &.-size-s {
        height: auto;
        min-height: 24px;
        padding:0 8px;
        font-size: 12px;
        .ma-btn-content {
          padding: 2px 0;
        }
        .translatedKw {
          font-size: 0.70rem;
        }
      }
      .ma-btn-icon {
        margin-top: 4px;
      }
    }
</style>
