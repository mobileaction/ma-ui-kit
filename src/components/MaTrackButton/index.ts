import MaTrackButton from './MaTrackButton.vue';

export default MaTrackButton;
