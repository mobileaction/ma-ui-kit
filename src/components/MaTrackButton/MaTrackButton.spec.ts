import { mount, Wrapper } from '@vue/test-utils';
import MaTrackButton from './';
import Vue from 'vue'

describe('MaTrackButton', () => {
  let trackButtonWrapper: Wrapper<Vue>;
  let button:Wrapper<Vue>;

  beforeEach(() => {
    trackButtonWrapper = mount(MaTrackButton, {
      propsData: {
        keyword: 'calm',
        disabled: false,
      },
    });

    button = trackButtonWrapper.find("button");
  });

  it('is disabled', () => {
    expect(button.attributes('disabled')).toBeFalsy();
  });

  it('prints keyword to button', () => {
    expect(button.text()).toBe("calm")
  });

  it('triggers a click', async () => {
    await button.trigger('click');
    expect(trackButtonWrapper.emitted()).toHaveProperty('click');
  });

  it('emits click event', async () => {
    trackButtonWrapper.vm.$emit('click', 123);
    await button.trigger('click')
    expect(button.emitted().click[0]).toEqual([123])
  });
});
