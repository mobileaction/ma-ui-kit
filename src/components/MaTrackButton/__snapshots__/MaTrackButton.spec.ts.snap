// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaTrackButton returns the same output 1`] = `
<button type="button" class="ma-button ant-btn -size-m -"><span class="ma-btn-content items-center"><span class="flex flex-wrap content-center mr-1"><span><span><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="plus" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="ma-btn-icon svg-inline--fa fa-plus fa-1x ma-undefined"><path fill="currentColor" d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z" class=""></path></svg></span></span></span> <span class="text-left leading-tight"><p>
                cal
            </p> <!----></span></span></button>
`;
